Source/WebService/Blackjack.WebService/ClientApp/src/app/components/play-game-page/play-game.component.ts

```diff
@@ -27,21 +65,160 @@ export class PlayGameComponent implements OnInit {
+    }
+    this.deck.shuffle(decks);
+    this.player.updateHand(this.deck.deal(2));
+    this.dealer.updateHand(this.deck.deal(2));
+    this.isHidden = false;
+
+    /*Checks for blackjacks. If anyone does, game automatically ends.*/
+    if (this.player.hasBlackJack() && this.dealer.hasBlackJack()) {
+      this.revealDealer();
+      this.nextMove = BlackjackOutcomes.BothBlackjack;
+      this.pushOutcome();
+      this.isPlayAgainHidden = false;
+      return;
+    }
+    else if (this.player.hasBlackJack() && !this.dealer.hasBlackJack()) {
+      this.revealDealer();
+      this.nextMove = BlackjackOutcomes.PlayerBlackjack;
+      this.winOutcome();
+      this.isPlayAgainHidden = false;
+      return;
+    }
+    else if (!this.player.hasBlackJack() && this.dealer.hasBlackJack()) {
+      this.revealDealer();
+      this.nextMove = BlackjackOutcomes.DealerBlackjack;
+      this.loseOutcome();
+      this.isPlayAgainHidden = false;
+      return;
+    }
+    this.nextMove = BlackjackOutcomes.PlayerChoice;
+    this.isGameButtonHidden = false;
+  }
+
+  /**
+   * Gives the player another card but automatically ends play if the player busts. 
+   */
+  public hit(): void {
+    this.player.updateHand(this.deck.deal(1));
+    if (this.player.hasBusted) {
+      this.revealDealer();
+      this.nextMove = BlackjackOutcomes.DealerWin;
+      this.loseOutcome();
+      this.isPlayAgainHidden = false;
+      this.isGameButtonHidden = true;
+    }
   }
 
+  /**
+   * Ends the players turn, reveals dealer's cards, hides game buttons, and plays
+   * the dealer's hand over two second intervals.
+   */
+  public stay(): void {
+    this.revealDealer();
+    const intervalID = setInterval(() => {
+      if (this.dealer.score > 17 || (this.dealer.score === 17 && !this.dealer.hasSoft17()) || this.dealer.hasBusted) {
+        clearInterval(intervalID);
+        if (this.dealer.score < this.player.score || this.dealer.hasBusted) {
+          this.nextMove = BlackjackOutcomes.PlayerWin;
+          this.winOutcome();
+        }
+        else if (this.dealer.score > this.player.score) {
+          this.nextMove = BlackjackOutcomes.Push;
+          this.pushOutcome();
+        }
+        else {
+          this.nextMove = BlackjackOutcomes.DealerWin;
+          this.loseOutcome();
+        }
+        this.isPlayAgainHidden = false;
+        return;
+      }
+      this.dealer.updateHand(this.deck.deal(1));
+    }, 2000);
+  }
+
+  /**
+   * Resets the player and dealer and starts a new game.
+   */
+  public playAgain(): void {
+    this.isHidden = true;
+    this.isPlayAgainHidden = true;
+    this.isGameButtonHidden = true;
+    this.player.reset();
+    this.dealer.reset();
+  }
 
+  /**
+   * Changes the dealer state to reveal his cards and prevent player from taking actions. 
+   */
+  public revealDealer(): void {
+    this.dealer.isDealerTurn = true;
+    this.dealer.updateHand(this.deck.deal(0));
```

Comment:
What's the reason for this line?

Reply:
I made the getStatus (haven't changed the name yet) method private but invoked in updateHand so I needed a way to trigger the change in status for the dealer. I could make the method public

Reply:
In this case I would add a new publicly accessible method like,  "recalculateStatus" or something. 

As it is right now you're taking advantage of the fact that you know how something works under the hood and it has lead to code that doesn't make sense to someone,  "outside of the know" and isn't intuitive. 

Additionally because updateHand does multiple things your objective is not inherently clear.

As well as if anything in updateHand changed, invoking this here could have unintended consequences in the future. 

Creating something contextually-purpose-built will safeguard against these things.